cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -313,6 +296,38 @@ func start(log log.Component,
 		return err
 	}
 
+	clusterName := clustername.GetRFC1123CompliantClusterName(context.TODO(), hname)
+	// Generate and persist a cluster ID
+	// this must be a UUID, and ideally be stable for the lifetime of a cluster,
+	// so we store it in a configmap that we try and read before generating a new one.
+	coreClient := apiCl.Cl.CoreV1().(*corev1.CoreV1Client)
+	//nolint:revive // TODO(CINT) Fix revive linter
+	clusterId, err := apicommon.GetOrCreateClusterID(coreClient)
+	if err != nil {
+		pkglog.Errorf("Failed to generate or retrieve the cluster ID")
+	}
+
+	if clusterName == "" {
+		pkglog.Warn("Failed to auto-detect a Kubernetes cluster name. We recommend you set it manually via the cluster_name config option")
+	}
+
+	// Initialize and start remote configuration client
+	var rcClient *rcclient.Client
+	rcserv, isSet := rcService.Get()
+	if pkgconfig.IsRemoteConfigEnabled(config) && isSet {
+		var err error
+		rcClient, err = initializeRemoteConfigClient(mainCtx, rcserv, config, clusterId, clusterName)
```

Comment:
❓ 

Should we reach this point if `clusterId` and/or `clusterName` is/are empty?